extension/ringmembershipextension/ringmembershipextension.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ringmembershipextension
+
+import (
+	"encoding/json"
+	"net"
+	"net/http"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+
+	"github.com/facette/natsort"
+	"github.com/open-telemetry/opentelemetry-collector/component"
+	"github.com/open-telemetry/opentelemetry-collector/extension"
+)
+
+type ringMembershipExtension struct {
```

Comment:
What is the purpose of this extension? This lacks documentation and a README.

It appears that all it does is queries a DNS and serves the resolved IP list via HTTP. Does it need to be a separate extension if so? Justification for having the extension is not clear.

Reply:
As mentioned in the design doc, a first pass minimum viable solution is to use DNS, which can be extended to include other service discovery mechanisms like an etcd client / consul etc.

Reply:
Secondly, an extension is useful because if multiple components require service discovery, they can rely on the extension instead of having to independently implement it.